src/utils/events/event-manager.js

```diff
@@ -117,26 +125,35 @@ export default class EventManager {
    */
   off(event, handler) {
     if (typeof event === 'string') {
-      // Clean up aliased gesture handler as necessary.
-      const recognizerEvent = EVENT_RECOGNIZER_MAP[event];
-      if (recognizerEvent) {
-        const eventAlias = GESTURE_EVENT_ALIASES[event];
-        if (eventAlias && this.aliasedEventHandlers[event]) {
-          this.manager.off(eventAlias, this.aliasedEventHandlers[event]);
-          delete this.aliasedEventHandlers[event];
-        }
-      }
-
-      // Deregister event handler.
-      this.manager.off(event, handler);
+      this._offOne(event, handler);
     } else {
       // If `event` is a map, call `off()` for each entry.
       for (const eventName in event) {
-        this.off(eventName, event[eventName]);
+        this.offOne(eventName, event[eventName]);
       }
     }
   }
 
+  _offOne(event, handler) {
+    // Clean up aliased gesture handler as necessary.
+    const recognizerEvent = EVENT_RECOGNIZER_MAP[event];
+    if (recognizerEvent) {
+      const eventAlias = GESTURE_EVENT_ALIASES[event];
+      if (eventAlias && this.aliasedEventHandlerCounts[event]) {
+        if (--this.aliasedEventHandlerCounts[event] <= 0) {
+          // If no aliased handlers remaining for this event,
+          // remove the aliased handler from Hammer.
+          // (The original handler is removed below.)
+          this.manager.off(eventAlias);
+          delete this.aliasedEventHandlerCounts[event];
+        }
+      }
+    }
+
+    // Deregister event handler.
+    this.manager.off(event, handler);
```

Comment:
For posterity:

`aliasedEventHandlerCounts[event]` only gets decremented if it's truthy, and if the handler wasn't previously registered and aliased it won't be truthy.

also, i already have a reset in there:
`delete this.aliasedEventHandlerCounts[event];`